extensions/mssql/package.json

```diff
@@ -557,7 +557,13 @@
           "description": "%mssql.connectionOptions.authType.description%",
           "groupName": "Security",
           "valueType": "category",
-          "defaultValue": null,
+          "defaultValue": "SqlLogin",
```

Comment:
why are we doing this?

Reply:
We are doing this because the default value is currently hard-coded in the connection widget to be "Integrated" if the OS is Windows. However, for PostgreSQL (and later MySQL), there is no "Integrated" authType. There is only "SqlLogin" (username/password), and "AzureMFAAndUser" (active directory). When the hard-coded value "Integrated" is assigned as the selected value of the authentication type dropdown that does not contain that value, things break in the connection dialog.

Reply:
Let me know if this answer is acceptable to you, or if you think there is a more straightforward way to deal with this. I will leave this thread open for a day or so or until I hear back from you. Thanks.

Reply:
I'm going to make one more comment on this, which is that the alternative to this would be to just change the original logic in connectionWidget.ts to set the default to "Integrated" when OS is Windows _and_ provider is either mssql or cms, otherwise set to "SqlLogin". This approach involves fewer code changes, but it hard-codes two provider names, two authTypes, and one operating system into connectionWidget.ts. I really feel like it is better to make this data-driven in package.json. Please re-open if you disagree and want to discuss it.